src/loader/SpineLoader.ts

```diff
@@ -89,128 +85,128 @@ export class SpineLoader extends Loader<SpineResource> {
-        resource = await this._handleOriginAsset(item, resourceManager);
+      let { fileExtensions } = item.params || {};
+      const spineAssetPath: SpineAssetPath = {
+        skeletonPath: '',
+        atlasPath: '',
+        imagePaths: [],
+        imageExtensions: [],
+      };
+      this._isSingleUrl = !item.urls;
+      if (this._isSingleUrl) {
+        const fileExtension = SpineLoader.verifyFileExtensions(fileExtensions, false);
+        SpineLoader.deriveAndAssignSpineAsset(item.url, fileExtension as string, spineAssetPath);
       } else {
-        const buffer: ArrayBuffer = await this.request(item.url, { type: 'arraybuffer' });
-        const reader = new BufferReader(new Uint8Array(buffer));
-        const header = reader.nextStr();
-        if (header.startsWith('spine')) {
-          resource = await this._handleEditorAsset(item, buffer, reader, header, resourceManager);
-        } else {
-          resource = await this._handleOriginAsset(item, resourceManager, buffer);
+        fileExtensions = SpineLoader.verifyFileExtensions(fileExtensions, true);
+        const urls = item.urls;
+        const urlsLen = urls.length;
+        for (let i = 0; i < urlsLen; i += 1) {
+          const url = urls[i];
+          const extension = fileExtensions && fileExtensions[i] || null;
+          SpineLoader.parseAndAssignSpineAsset(url, extension, spineAssetPath);
+        }
+      }
+
+      const { skeletonPath, atlasPath } = spineAssetPath;
+      if (!skeletonPath || !atlasPath) {
+        throw new Error('Failed to load spine assets. Please check the file path and ensure the file extension is included.');
+      }
+
+      this._fileName = this._extractFileName(skeletonPath);
+
+      let skeletonRawData: ArrayBuffer | string = await this.request(skeletonPath, { type: 'arraybuffer' }) as ArrayBuffer;
+      this._bufferReader = new BufferReader(new Uint8Array(skeletonRawData));
+      let isEditorAsset = false;
+      try {
+        const decoder = this._decoder;
+        const jsonString = decoder.decode(skeletonRawData);
+        const parsedJson = JSON.parse(jsonString);
+        skeletonRawData = jsonString;
+        if (parsedJson.spine) {
+          isEditorAsset = true;
         }
+      } catch {
+        
+        try {
+          this._bufferReader = new BufferReader(new Uint8Array(skeletonRawData as ArrayBuffer));
+          const header = this._bufferReader.nextStr();
+          isEditorAsset = header.startsWith('spine');
+        } catch {} // origin asset may exceed when call next str 
+      }
+
+      if (!this._isSingleUrl) {
+        // origin asset
+        resource = await this._handleOriginAsset(skeletonRawData, spineAssetPath);
+      } else if (isEditorAsset) {
+        // editor asset
+        resource = await this._handleEditorAsset(skeletonRawData);
+      } else {
+        // origin asset
+        resource = await this._handleOriginAsset(skeletonRawData, spineAssetPath);
       }
       resolve(resource);
```

Comment:
Only when it is a single file, it needs to be determined whether it is an editor asset